frontend/src/hooks/useVerifyBoardEffect.ts

```diff
@@ -1,44 +1,183 @@
+    },
+    {
+      key: 'Site',
+      name: 'site',
+      value: boardFields.site,
+      defaultValue: EMPTY_STRING,
+      isRequired: true,
+      isValid: true,
+      errorMessage: '',
+      col: 1,
+    },
+    {
+      key: 'Token',
+      name: 'token',
+      value: boardFields.token,
+      defaultValue: EMPTY_STRING,
+      isRequired: true,
+      isValid: true,
+      validRule: (value: string) => REGEX.BOARD_TOKEN.test(value),
+      errorMessage: '',
+      col: 2,
+    },
+  ]);
+
+  const resetFormFields = () =>
+    setFormFields(
+      formFields.map((field) => {
+        return { ...field, value: EMPTY_STRING, isRequired: true, isValid: true };
+      }),
+    );
+
+  const clearError = () => {
+    return setFormFields(
+      formFields.map((item) => ({
+        ...item,
+        isValid: true,
+        isRequired: true,
+        errorMessage: '',
+      })),
+    );
+  };
+
+  const setErrorField = (names: string[], messages: string[]) => {
+    setFormFields(
+      formFields.map((field) => {
+        return names.includes(field.name)
+          ? { ...field, isValid: false, errorMessage: messages[names.findIndex((name) => name === field.name)] }
+          : field;
+      }),
+    );
+  };
+
+  const validField = (field: FormField, inputValue: string) => {
+    const value = inputValue.trim();
+    const isRequired = !!value;
+    const isValid = !field.validRule || field.validRule(field.value.trim());
+    const errorMessage = !isRequired
+      ? `${field.key} is required`
+      : !isValid
+        ? `${field.key} is invalid`
+        : DEFAULT_HELPER_TEXT;
+
+    return {
+      ...field,
+      value,
+      isRequired,
+      isValid,
+      errorMessage,
+    };
+  };
+
+  const updateField = (name: string, value: string) => {
+    setFormFields(
+      formFields.map((field) => {
+        return field.name === name ? validField(field, value) : field;
+      }),
+    );
+  };
 
-  const verifyJira = async (params: BoardRequestDTO) => {
+  const verifyJira = (params: BoardRequestDTO) => {
     setIsLoading(true);
-    try {
-      return await boardClient.getVerifyBoard(params);
-    } catch (e) {
-      const err = e as Error;
-      setErrorMessage(`${params.type} ${MESSAGE.VERIFY_FAILED_ERROR}: ${err.message}`);
-      setTimeout(() => {
-        setErrorMessage('');
-      }, DURATION.ERROR_MESSAGE_TIME);
-    } finally {
-      setIsLoading(false);
-    }
+    return boardClient
+      .getVerifyBoard({
+        ...params,
+        token: getJiraBoardToken(params.token, params.email),
+      })
+      .then((result) => {
+        clearError();
+        return result;
+      })
+      .catch((e) => {
+        const { message, code } = e;
+        if (code === HttpStatusCode.Unauthorized) {
+          setErrorField(['email', 'token'], [MESSAGE.VERIFY_MAIL_FAILED_ERROR, MESSAGE.VERIFY_TOKEN_FAILED_ERROR]);
+        }
+        if (code === HttpStatusCode.NotFound && message === 'site not found') {
```

Comment:
extract message as const or object mapping

Reply:
do not proposal for use， error is in a different hierarchy